### DISCUSSION TITLE
LinkedIn Login always shows login even if already logged

### DISCUSSION THREAD

**Discussion by sourcecodeit** [supabase/supabase#10681] [Questions]
Hi all, I'm adding LinkedIn auth in my Vue 3 web app. It works correctly but it always shows the LinkedIn login page even if the user is already logged.

I'm currently using the basic setup:

```
const supabase = createClient(import.meta.env.VITE_SUPABASE_URL, import.meta.env.VITE_SUPABASE_KEY);

async function loginInLinkedIn() {
	const { data, error } = await supabase.auth.signInWithOAuth({
  provider: 'linkedin'
});
```

Do you have any idea about the reason and how to solve this?

**Comment by silentworks** [comment]
Can you show your full code please? This bit of code won't be enough to let us know why you are still seeing the login page. Where is the rest of the code for your login page? are you using a router? is this a SPA without a router? can you see the `[projectref]-auth-token` in your localSTorage?

**Reply by sourcecodeit** [reply]
Sorry I didn't describe the issue very well. My auth flow works seamlessly:

1. I call `supabase.auth.signInWithOAuth({ provider: 'linkedin' });`
2. I login into LinkedIn
3. I'm redirected to my web app
4. I get my token in onAuthStateChange()

The problem is at point 2. because even if I'm already logged in LinkedIn calling `supabase.auth.signInWithOAuth({ provider: 'linkedin' });` will keep show Login page of LinkedIn.

The LinkedIn login page is: https://www.linkedin.com/uas/login (followed by params like session_redirect, app_id, etc)

I guess the correct flow shouldn't show the LinkedIn's login page, if the user is already logged in LinkedIn. 

Not sure if this it clearer now :)

**Reply by silentworks** [reply]
My question is where is the code in your web app that is supposed to stop the user from seeing the login page again? This is logic you will have to write, its not something the supabase library will or should do for you. The library is basically just setting the state of the user [signed_in/not_signed_in]. Can you share your code where you are handling the state of your user?

**Reply by sourcecodeit** [reply]
Ok this is my code but this code is working, I can't say in other words.

**App.vue**
```
authStore.supabase.auth.onAuthStateChange((event, session) => {
  if(session.user) {
    authStore.user = session.user;
    authStore.access_token = session.access_token;

    // set the cookie for the chrome extension
    cookies.set("access_token", session.access_token);
  }
})
```

**router.js**
```
router.beforeEach((to, from, next) => {
  const publicPages = ["/login", "/access_token"];
  const authRequired = !publicPages.includes(to.path);
  const auth = useAuthStore();
  if (authRequired && !auth.access_token) {
    auth.returnUrl = to.fullPath;
    next({
      name: "login",
    });
    return;
  }
  next();
});
```

**authStore.js**
```
  actions: {
    async loginWithLinkedIn() {
		  const { data, error } = await this.supabase.auth.signInWithOAuth({
			  provider: 'linkedin'
		  });
    },
    logout() {
      this.user = null;
      cookies.remove('access_token');
      this.supabase.auth.signOut();
      router.push("/login");
    },
  },
```

**LoginView.vue**
```
<BaseButton class="w-full" color="info" label="Connect with LinkedIn" :rounded-full="true"
              @click="authStore.loginWithLinkedIn()" />
```

After logging in with LinkedIn I don't see my login page anymore, **that's just perfect**. The problem is with LinkedIn. If I log out my web app, then I click again my login button I keep seeing the **LinkedIn** login page. Why? I didn't log out from LinkedIn.

**Comment by jibbolo** [comment]
+1
This is happening to me too. With google as a login provider, I get logged in automatically without explicitly signing in. With Linkedin instead, I always see the login form, no matter if I was already signed in. I don't know if this is a Linkedin limitation or not, I just think it's not the best UX for the user.
